drivers/tee/optee/optee.c

```diff
@@ -747,30 +790,141 @@ static int optee_get_version(const struct device *dev, struct tee_version_info *
 	return 0;
 }
 
+static size_t get_shm_size(const struct device *dev)
+{
+	struct optee_driver_data *data = (struct optee_driver_data *)dev->data;
+	size_t shm_size = OPTEE_MSG_GET_ARG_SIZE(MAX_ARG_PARAM_COUNT);
+
+	if (data->rpc_param_count != 0) {
+		shm_size += OPTEE_MSG_GET_ARG_SIZE(data->rpc_param_count);
+	}
+
+	return shm_size;
+}
+
+static int alloc_shm_cache_entry(struct shm_cache_entry **entry)
+{
+	if (entry == NULL) {
+		return -EINVAL;
+	}
+
+	*entry = k_malloc(sizeof(**entry));
+	if (*entry == NULL) {
+		return -ENOMEM;
+	}
+
+	memset(*entry, 0, sizeof(**entry));
+
+	return 0;
+}
+
+#if defined(CONFIG_OPTEE_CLEAN_SHM_CACHE)
+static void rm_shm_if_empty_unlocked(const struct device *dev, struct shm_cache_entry *entry)
```

Comment:
function called `rm if empty`  but it calls `tee_rm_shm` unconditionally. I am not sure that I got the logic behind this